### DISCUSSION TITLE
keyball44 トラックボールが反応しない

### DISCUSSION THREAD

**Discussion by boxwood8** [Yowkees/keyball#403] [Q&A]
お世話になっております。
先日トラシューのためトラックボール基板を追加購入させていただき取り付けましたがボールが反応しません。OLED 上のセンサーの値は0から変化無しです。キーは全て反応しています。

以下切り分け済みです
- 組み立て済みトラックボール基板に交換
- Shirogane Lab ショップより購入した Type-C Pro Micro に再度交換
- ジャンパのハンダ付けチェック
- トラックボール基板接触不良疑いのため7Pソケットへのハンダ付け
- 7P ソケットとミドル基板 ProMicro コンスルー間の導通チェック

ProMicro との導通チェックを行いたいのですがピンの組み合わせをご教授ください。
また最新のファームウェアと ProMicroとの相性問題など報告があれば教えて下さい

![image](https://github.com/Yowkees/keyball/assets/147312996/7a66ca10-a550-4fca-8e8e-f45600fe9f49)
![image](https://github.com/Yowkees/keyball/assets/147312996/0bd146b7-5e57-4a49-a3a7-22d648599399)

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

ご使用されているProMicroでしたら問題なく動作します。

お写真を確認すると、半田の量が少なく導通不良となりそうな個所があります。
下画像の丸をつけた端子に半田を足して、半田が山型になるように修正お願いします。

その他、トラックボール基板でも半田が少なそうな個所がないかご確認ください。

おそらく半田を足すことで動作すると思います。お手数ですがご確認お願いいたします。

![ProMicro_Fix](https://github.com/Yowkees/keyball/assets/84209772/5456529b-2319-460c-a111-d5744bf160c9)

**Comment by Yowkees** [comment]
導通チェックについては、下記にKeyball44の回路図と基板データを公開しています。
https://github.com/Yowkees/keyball/tree/main/keyball44/design_data

KiCAD等のフリーソフトで表示していただき、線を追っていくとどこが繋がっているか確認いただけると思います。
お手数ですがご確認お願いいたします。

**Reply by boxwood8** [reply]
早速のご対応ありがとうございました

確認してみます

**Reply by boxwood8** [reply]
お世話になっております。

ハンダ付けを見直しましたが残念ながら解消しませんでした。
写真の箇所についてハンダ上の接点同士はテスターでは導通しておりますので、何か根本的な勘違いがあるのかもしれません…
OLED 上のボールのトラッキング量の表示はボール操作のみで本来反応するものでしょうか？

![image](https://github.com/Yowkees/keyball/assets/147312996/779ff323-57f2-468f-ba37-434d15b23136)

**Comment by Yowkees** [comment]
導通確認ありがとうございます。

OLEDの表示は、ボールを動かすだけで数値が動きます。
気になったのは、光学センサーがトラックボール基板に密着せず少し浮いた状態になっているかなと感じました。
最初のメッセージに「組み立て済みトラックボール基板に交換」とありますが、これは白銀ラボでご購入されたものでしょうか？

L字7Pコンスルーのトラックボール基板側は半田付けの必要はありませんが、半田付けされているように見えますので、この周辺に半田クズが飛んでいないかご確認ください。トラックボール基板を取り外せそうでしたら、これを新しいものに交換してしまうのも手だと思います。

以上、お手数ですがご確認お願いいたします。

**Reply by OTR7190** [reply]
横から失礼します。

私の場合はノートPC用のドッキングステーション経由でUSB接続しているとトラックボールが動かない事象が発生しました。
ノートPC直差しで運用しております。

同じような事が無いか気になったのでご確認ください。

**Reply by boxwood8** [reply]
ありがとうございます。

_最初のメッセージに「組み立て済みトラックボール基板に交換」とありますが、これは白銀ラボでご購入されたものでしょうか？_
はい、前回のトラシューでミドル基板側に問題無さそうと判断し、ProMicro と共に白銀ラボさんのストアから購入しました。

PCとの接続には確かにドックを経由してましたので直結に変更しましたが残念ながら解消せずです。

44キーでの作業に手も慣れてきたので勉強代と諦めようかと思います

**Comment by Yowkees** [comment]
[boxwood8](https://github.com/boxwood8)様

半田付け＆動作確認済のトラックボール基板でも動作しなかったとの事で了解しました。
今後の対応方法として、当方での実機確認を含めてお話ししたいので、一度、下記白銀ラボのお問合せからご連絡いただけますでしょうか。
https://shirogane-lab.net/inquiry

お手数ですがご対応よろしくお願いいたします。

**Reply by boxwood8** [reply]
承知いたしました。
お手数ですがその際はよろしくお願いします。
